### PR TITLE
@task PDA-1223 GET /v1/cartItems - use GraphQL query in front end

### PR DESCRIPTION
**Pull request by lli-parasoft**

### CODE REVIEW

**Review by rwang-parasoft — APPROVED** [review]

**Comment by yshi-parasoft** [comment]
`graphQLService.js` should be embedded into `orders.html` or we'll get console error when initializing `requisitionBarController` in orders page.
image

File: src/main/resources/static/common/js/graphQLService.js [hunk 1]
```diff
@@ -45,5 +45,12 @@ angular
                     success(response.data.data.getLocation);
                 }, error);
             }
+            //cartItems
+            instance.getCartItems = function(success, error) {
+                let requestBody = {"query": "query{getCartItems{id,userId,itemId,name,description,image,realInStock,quantity}}"}
```
Comment (yshi-parasoft) [inline src/main/resources/static/common/js/graphQLService.js:50]: Let's change the arguments to be passed in argument. And for all places which make this query, only select fields needed to minimize payload. That's the purpose of GraphQL.

**Comment by yshi-parasoft** [comment]
Looks like we need to select more fields here.
image

**Comment by yshi-parasoft** [comment]
Please review all places using this common method "loadShoppingCartData" see if we need to load everything for this case. 
My understanding is that we may only need the count number for displaying it on cart icon.

**Comment by yshi-parasoft** [comment]
The current selectionSet is still incorrect. According to the front end code, what we need are what's used from line 141 to line 162 for "cartItem". 
image

Your current implementation looks right because $scope.cartItems get set earlier. But if you add more item to cart from another browser tab before going to order details page, you won't see the image of the newly added item in cart.
image

**Pull request merged**